tests/blocks/test_notifications.py

```diff
@@ -367,3 +368,175 @@ def test_invalid_to_phone_numbers_raises_warning(self, caplog):
             )
 
             assert "Dropped invalid phone # (0000000) specified." in caplog.text
+
+
+class HttpxResponseMock:
+    def __init__(self) -> None:
+        self.raise_for_status = Mock()
+
+
+class TestCustomWebhook:
+    async def _test_notify_async(
+        self,
+        data: dict,
+        expected_call: dict,
+        body: str = "test",
+        subject: Optional[str] = "subject",
+    ):
+        with patch("httpx.AsyncClient", autospec=True) as HttpxClientMock:
```

Comment:
You should be able to to [`respx` ](https://lundberg.github.io/respx/)for mocking out network calls via `httpx`.